scopes/component/component/component-fs.ts

```diff
@@ -34,12 +34,22 @@ export default class ComponentFS extends MemoryFS {
    * filter all files using an array of glob patterns.
    */
   byGlob(patterns: string[]) {
-    return this.files.filter((file) => {
-      return patterns.find((pattern) => {
+    const excludePatterns = patterns.filter((pattern) => pattern.startsWith('!'));
+    const includePatterns = patterns.filter((pattern) => !pattern.startsWith('!'));
+
+    const files = this.files.filter((file) => {
+      const matchesSome = includePatterns.some((pattern) => {
```

Comment:
rename - it should somehow represent that it's based on the included
maybe call it just `included`